### ISSUE TITLE
Unexpected data transformation when transformation in formula

### ISSUE THREAD

**Issue by DominiqueMakowski** [easystats/insight#639]
When a variable is of the form `log(1 + var)` in the formula, it transforms *other* variables when we get the data... very strange

``` r
summary(iris$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   4.300   5.100   5.800   5.843   6.400   7.900


model <- lm(Petal.Length ~ log(Sepal.Width) + Sepal.Length,
            data=iris)

# Same min-max
summary(insight::get_data(model)$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   4.300   5.100   5.800   5.843   6.400   7.900

model <- lm(Petal.Length ~ log(1 + Sepal.Width) + Sepal.Length,
            data=iris)

# Diffrent!
summary(insight::get_data(model)$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   3.000   3.800   4.000   4.057   4.300   5.400
```

<sup>Created on 2022-09-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by strengejacke** [comment]
What happens if you use `log1p()`?

**Comment by DominiqueMakowski** [comment]
then it works

**Comment by strengejacke** [comment]
should be fixed in #640

**Comment by strengejacke** [comment]
``` r
summary(iris$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   4.300   5.100   5.800   5.843   6.400   7.900
model <- lm(Petal.Length ~ log(Sepal.Width) + Sepal.Length,
  data = iris
)
# Same min-max
summary(insight::get_data(model)$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   4.300   5.100   5.800   5.843   6.400   7.900

model <- lm(Petal.Length ~ log(1 + Sepal.Width) + Sepal.Length,
  data = iris
)
summary(insight::get_data(model)$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   4.300   5.100   5.800   5.843   6.400   7.900

model <- lm(Petal.Length ~ log(Sepal.Width + 1) + Sepal.Length,
  data = iris
)
summary(insight::get_data(model)$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   4.300   5.100   5.800   5.843   6.400   7.900

model <- lm(Petal.Length ~ log1p(Sepal.Width) + Sepal.Length,
  data = iris
)
summary(insight::get_data(model)$Sepal.Length)
#>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#>   4.300   5.100   5.800   5.843   6.400   7.900
```

<sup>Created on 2022-09-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
